docs/DataRequirements.md

```diff
@@ -0,0 +1,252 @@
+  catching up the relayer to the latest executed DA block and for status checks.
+
+- `/prover_aggregatedProofsData`: Used to create update client messages,
+  specifically in situations where updating a client for a range of slots or
+  heights surpasses the coverage provided by a single proof, therefore it
+  requires the submission of a list of aggregated proofs.
+
+- status: Nothing available yet.
+
+### `/rollup_params`
+
+- Used for adjusting rollup parameters that the relayer may need for setting
+  configurations and basic checks, like the namespaces, max batch size, max tx
+  size, max gas fees, etc.
+- Used to retrieve the rollup code commitment, essential for the aggregated
+  proof verification.
+- Only used once, at relayer startup during health check.
+- Usually not needed for IBC relaying strictly speaking.
+
+- Status: Nothing available yet.
+
+### `/status`
+
+- Needed to get rollup status including node info, pubkey, latest DA block hash,
+  rollup state root, height (slot number) and time.
+- Assuming the `/status` returns similar
+  [`Response`](https://github.com/informalsystems/tendermint-rs/blob/main/rpc/src/endpoint/status.rs#L26-L37)
+  type to Cosmos chains, The response used in two situations:
+  1. At relayer startup to fetch `node_info.id`, for initializing the light
+     client component.
+      - Also at startup to fetch `node_info.other.tx_index`, during health
+        check.
+      - Also at startup to fetch `node_info.network`, i.e., the network
+        identifier, during health check.
+      - Also at startup to assert that `sync_info.catching_up` is false, during
+        health check.
+  2. To fetch the rollup latest time and height used in many methods, often
+    alongside `node_info.network`, for example:
+      - As a dependency, because the latest height is necessary in calling the
+        `/rollup_params` RPC, during health check.
+      - Needed in channel handshake (open try, ack, confirm; close confirm).
+      - In connection handshake (open try, ack, confirm), for both the source
+        chain (to update client), and destination chain (to construct proofs).
+      - Note: It seems like we bombard the node with `/status` queries, but most
+        of the queries hit the Hermes in-process cache.
+      - For updating clients during everyday IBC relaying. In case there is
+        non-zero connection delay, we again bombard the node with `/status`
+        queries.
+
+- Status: Nothing available yet, but part of the needed data can be gathered
+  from different endpoints like `ledger_getHead` or
+  `prover_AggregatedProofData`.
+
+### `/rollup_health`
+
+- Needed for basic check to assess the health of the rollup nodes.
+- Only used once, at relayer startup during health check.
+- Not needed for IBC relaying strictly speaking.
+- In case of Cosmos chains, it returns empty result (200 OK) on success.
+
+- Status: Available as a `/health` method on RPC client
+
```

Comment:
We also need to query the rollup for the jellyfish proof (we use `/abci_query` in cosmos chains) not sure if/ where this is covered.
In addition we need to be able to query the account and nonce (i guess to the sequencer) but not sure what types of accounts it uses and how they work.